crates/gitbutler-tauri/src/diff.rs

```diff
@@ -34,6 +38,63 @@ pub fn changes_in_commit(
         .map_err(Into::into)
 }
 
+#[tauri::command(async)]
+#[instrument(skip(projects, settings), err(Debug))]
+pub fn changes_in_branch(
+    projects: tauri::State<'_, gitbutler_project::Controller>,
+    settings: tauri::State<'_, but_settings::AppSettingsWithDiskSync>,
+    project_id: ProjectId,
+    stack_id: StackId,
+    branch_name: String,
+) -> anyhow::Result<Vec<TreeChange>, Error> {
+    let project = projects.get(project_id)?;
+    let ctx = CommandContext::open(&project, settings.get()?.clone())?;
+    changes_in_branch_inner(ctx, branch_name, stack_id).map_err(Into::into)
+}
+
+fn changes_in_branch_inner(
+    ctx: CommandContext,
+    branch_name: String,
+    stack_id: StackId,
```

Comment:
I think right now it's hard to *not* use `StackId` as there is no real V3 counterpart for dealing with stacks or learning about them.

The `head_info()` function, for instance, would naturally return with a deterministic list of commits per stack-segment, which is exactly what's needed here - all it wants is the top-most and bottom-most commit belonging to a stack.

On the bright side, once that becomes available it should be quite straightforward to find all places with StackId and let them use the new APIs.